cli/src/pcluster/resources/compute_node/user_data.sh

```diff
@@ -102,84 +102,106 @@ write_files:
+          if [[ "${!bucket_region}" == null ]]; then
+            bucket_region="us-east-1"
+          fi
+          cookbook_url=$(aws s3 presign "${!custom_cookbook}" --region "${!bucket_region}")
+        else
+          cookbook_url=${!custom_cookbook}
+        fi
+      fi
+      export PATH=/usr/local/sbin:/usr/local/bin:/sbin:/bin:/usr/sbin:/usr/bin:/opt/aws/bin
+      export parallelcluster_version=aws-parallelcluster-${ParallelClusterVersion}
+      export cookbook_version=${CookbookVersion}
+      export chef_version=${ChefVersion}
+      export berkshelf_version=${BerkshelfVersion}
+      if [ -f /opt/parallelcluster/.bootstrapped ]; then
+        installed_version=$(cat /opt/parallelcluster/.bootstrapped)
+        if [ "${!cookbook_version}" != "${!installed_version}" ]; then
+          error_exit "This AMI was created with ${!installed_version}, but is trying to be used with ${!cookbook_version}. Please either use an AMI created with ${!cookbook_version} or change your ParallelCluster to ${!installed_version}"
+        fi
+      else
+        error_exit "This AMI was not baked by ParallelCluster. Please use pcluster createami command to create an AMI by providing your AMI as parent image."
+      fi
+      if [ "${!custom_cookbook}" != "NONE" ]; then
+        curl --retry 3 -v -L -o /etc/chef/aws-parallelcluster-cookbook.tgz ${!cookbook_url}
+        vendor_cookbook
+      fi
+      cd /tmp
+
+      mkdir -p /etc/chef/ohai/hints
+      touch /etc/chef/ohai/hints/ec2.json
+      jq --argfile f1 /tmp/dna.json --argfile f2 /tmp/extra.json -n '$f1 * $f2' > /etc/chef/dna.json || ( echo "jq not installed or invalid extra_json"; cp /tmp/dna.json /etc/chef/dna.json)
+      {
+        pushd /etc/chef &&
+        cinc-client --local-mode --config /etc/chef/client.rb --log_level info --force-formatter --no-color --chef-zero-port 8889 --json-attributes /etc/chef/dna.json --override-runlist aws-parallelcluster::init &&
+        /opt/parallelcluster/scripts/fetch_and_run -preinstall &&
+        cinc-client --local-mode --config /etc/chef/client.rb --log_level info --force-formatter --no-color --chef-zero-port 8889 --json-attributes /etc/chef/dna.json --override-runlist aws-parallelcluster::config &&
+        /opt/parallelcluster/scripts/fetch_and_run -postinstall &&
+        cinc-client --local-mode --config /etc/chef/client.rb --log_level info --force-formatter --no-color --chef-zero-port 8889 --json-attributes /etc/chef/dna.json --override-runlist aws-parallelcluster::finalize &&
+        popd
+      } || error_exit 'Failed to run bootstrap recipes. If --norollback was specified, check /var/log/cfn-init.log and /var/log/cloud-init-output.log.'
+
+      if [ ! -f /opt/parallelcluster/.bootstrapped ]; then
+        echo ${!cookbook_version} | tee /opt/parallelcluster/.bootstrapped
+      fi
 
 --==BOUNDARY==
 Content-Type: text/x-shellscript; charset="us-ascii"
 MIME-Version: 1.0
 
 #!/bin/bash -x
 
+chmod +x /tmp/bootstrap.sh
```

Comment:
what if we set file permissions above to `700` and remove this chmod?

Reply:
Changed